src/Raven.Server/Documents/ETL/Providers/AI/GenAi/GenAiTask.cs

```diff
@@ -0,0 +1,392 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Raven.Client.Documents.Operations.AI;
+using Raven.Client.Documents.Operations.Counters;
+using Raven.Client.Documents.Operations.ETL;
+using Raven.Server.Documents.AI.AiGen;
+using Raven.Server.Documents.ETL.Metrics;
+using Raven.Server.Documents.ETL.Providers.AI.Embeddings;
+using Raven.Server.Documents.ETL.Providers.AI.Enumerators;
+using Raven.Server.Documents.ETL.Providers.AI.GenAi.Test;
+using Raven.Server.Documents.ETL.Stats;
+using Raven.Server.Documents.ETL.Test;
+using Raven.Server.Documents.Patch;
+using Raven.Server.Documents.Replication.ReplicationItems;
+using Raven.Server.Documents.TimeSeries;
+using Raven.Server.ServerWide;
+using Raven.Server.ServerWide.Context;
+using Raven.Server.Utils;
+using Sparrow.Json;
+using Sparrow.Json.Parsing;
+using Sparrow.Server.Utils;
+using Encoding = System.Text.Encoding;
+using PatchRequest = Raven.Server.Documents.Patch.PatchRequest;
+
+#pragma warning disable SKEXP0001
+
+namespace Raven.Server.Documents.ETL.Providers.AI.GenAi;
+
+public sealed class GenAiTask : EtlProcess<AiEtlItem, GenAiScriptResult, GenAiConfiguration, AiConnectionString,
+    GenAiStatsScope, GenAiPerformanceOperation>
+{
+
+    internal const string GenAiHashesMetadataKey = "@genAiHashes";
```

Comment:
It could be `@gen-ai-hashes` for consistency with other metadata that we already have: `@change-vector`, `@last-modified`, `@counters-snapshot`,  `@timeseries-named-values` etc

Please also consider moving it to `Raven.Client.Constants.Metadata`

Reply:
changed to `@gen-ai-hashes` and moved to `Raven.Client.Constants.Metadata`